services/ark-api/ark-api/src/ark_api/api/v1/openai.py

```
@@ -54,6 +57,62 @@ class ChatCompletionRequest(BaseModel):
     stream: bool = False
 
 
+async def try_get_streaming_url(ark_client, query_name: str, namespace: str = "default") -> Optional[str]:
+    """Try to get streaming URL, return None if memory doesn't support streaming."""
+    try:
+        # Get query to find memory reference
+        query = await ark_client.queries.a_get(query_name)
+        query_dict = query.to_dict()
+        
+        # Resolve memory name
+        memory_spec = query_dict.get("spec", {}).get("memory")
+        if memory_spec and memory_spec.get("name"):
+            memory_name = memory_spec["name"]
+        else:
+            memory_name = "default"
+            
+        # Try to get memory resource
+        memory = await ark_client.memories.a_get(memory_name)
+        memory_dict = memory.to_dict()
+        
+        # Check if memory supports streaming via annotation
+        annotations = memory_dict.get("metadata", {}).get("annotations", {})
+        streaming_enabled = annotations.get("ark.mckinsey.com/memory-event-stream-enabled") == "true"
+        
+        if not streaming_enabled:
+            return None
+        
+        # Check if memory has resolved address
+        status = memory_dict.get("status", {})
+        base_url = status.get("lastResolvedAddress")
+        
+        if not base_url:
+            return None
+        
+        # Use query name for streaming endpoint (query-based streaming)
+        # Return streaming URL with wait-for-query to handle query execution timing
+        return f"{base_url}/stream/{query_name}?from-beginning=true&wait-for-query=30s"
+        
+    except Exception:
+        # Any error means streaming not available
```

don't mask exceptions